diff --git a/source/spawn/spawn-processes__glib.adb b/source/spawn/spawn-processes__glib.adb
index 08e83ba..bd3fa4b 100644
--- a/source/spawn/spawn-processes__glib.adb
+++ b/source/spawn/spawn-processes__glib.adb
@@ -338,11 +338,12 @@ package body Spawn.Processes is
 
    overriding procedure Finalize (Self : in out Process) is
    begin
-      if Self.Status /= Not_Running then
+--      if Self.Status /= Not_Running then
 --           raise Program_Error;
          --  XXX commented out temporary to avoid finalization issues in GPS
-         null;
-      end if;
+--         null;
+--      end if;
+      null;
    end Finalize;
 
    --------------
