diff --git a/src_editor/src/src_editor_view.adb b/src_editor/src/src_editor_view.adb
index a04a81d..ac7b94a 100644
--- a/src_editor/src/src_editor_view.adb
+++ b/src_editor/src/src_editor_view.adb
@@ -1250,8 +1250,6 @@ package body Src_Editor_View is
             Draw_Below (Iter);
          when Text_View_Layer_Above =>
             Draw_Above;
-         when others =>
-            null;
       end case;
 
    exception
