diff --git a/prj_editor/src/project_properties.adb b/prj_editor/src/project_properties.adb
index cf02ce3..af52256 100644
--- a/prj_editor/src/project_properties.adb
+++ b/prj_editor/src/project_properties.adb
@@ -19,7 +19,6 @@ with Ada.Characters.Handling;      use Ada.Characters.Handling;
 with Ada.Strings.Unbounded;        use Ada.Strings.Unbounded;
 with Ada.Unchecked_Deallocation;
 
-with GNAT.Case_Util;
 with GNAT.Directory_Operations;    use GNAT.Directory_Operations;
 with GNAT.OS_Lib;                  use GNAT.OS_Lib;
 
