diff --git a/src/flow/flow_utility.adb b/src/flow/flow_utility.adb
index 862e6e7c..393d5037 100644
--- a/src/flow/flow_utility.adb
+++ b/src/flow/flow_utility.adb
@@ -2631,7 +2631,6 @@ function Do_Entity (E : Entity_Id) return Flow_Id_Sets.Set is
                raise Program_Error;
 
             when E_Package_Body
-               | E_Protected_Object
                | E_Protected_Body
                | E_Return_Statement
                | E_Subprogram_Body
diff --git a/src/spark/spark_definition.adb b/src/spark/spark_definition.adb
index bc537906..fafe7936 100644
--- a/src/spark/spark_definition.adb
+++ b/src/spark/spark_definition.adb
@@ -2394,6 +2394,7 @@ procedure Check_Raise_Context (Expr : Node_Id) is
             | N_Unused_At_End
             | N_Unused_At_Start
             | N_Variant
+            | N_Iterated_Element_Association
          =>
             raise Program_Error;
       end case;
@@ -6165,7 +6166,6 @@ procedure Mark_Identifier_Or_Expanded_Name (N : Node_Id) is
             | E_Block
             | E_Operator
             | E_Package_Body
-            | E_Protected_Object
             | E_Protected_Body
             | E_Subprogram_Body
             | E_Task_Body
